### PR TITLE
Update Operator Bundle Validate to use OperatorSdkEngine

### LINKED ISSUE #33: [Check] Operator Bundle Data Linting
Run `operator-sdk bundle validate -b podman --select-optional name=operatorhub --select-optional name=community registry.redhat.io/quay/quay-operator-bundle`  
Check should fail on errors and print to stdout and the log
Check should pass on warnings but print warnings to stdout and the log

### PR DESCRIPTION
**Pull request by bcrochet**
* Rename Policy to Check
* Add unit test for Check
* Fix a typo in BundleValidateOptions
* Add Verbose flag to BundleValidateOptions
* Add support for Verbose to OperatorSdkEngine implementation

Fixes #33

Signed-off-by: Brad P. Crochet <brad@redhat.com>

### CODE REVIEW

**Review by samira-barouti — APPROVED** [review]

File: certification/internal/shell/validate_operator_bundle.go
Comment (komish) [inline certification/internal/shell/validate_operator_bundle.go:9]: Nit: Same here re: check naming

File: certification/internal/shell/validate_operator_bundle.go
Comment (komish) [inline certification/internal/shell/validate_operator_bundle.go:53]: This should end with the `Check` suffix

File: certification/internal/shell/validate_operator_bundle.go
Comment (komish) [inline certification/internal/shell/validate_operator_bundle.go:47]: This is taking the output directly from `operator-sdk bundle validate` and mapping to our logging facilities in preflight, correct? This is a tricky approach for me; it directly ties the logging structure of operator-sdk to our logging structure. If their logging structure changes, we're unlikely to see it prior to shipping.

I know it's probably less pretty, but from the above perspective, would it be more stable to just pretty-print the entire log output at logging verbosity that we control so that the user has visibility into why something may have triggered a failure case? 

I do think we have this on other places as well. I'm not sure we need to solve it in this PR, but I did want to mention it to see what we could to give the user the needed visibility into things without hooking too deeply into logging behaviors which may not be reliable for our use case.

Also, if I misunderstood what this is doing, let me know/resolve away!

File: cli/operatorsdk.go
Comment (komish) [inline cli/operatorsdk.go:32]: Just want to confirm: this is here as a pass-through flag for operator-sdk, correct? We aren't exposing this to the user in any way, so this is _technically_ not configurable today, but may be something we can configure in the future?

**Review by komish — COMMENTED** [review]
Couple of questions/nits - no explicit change required for the moment.

File: certification/engine/engine.go
Comment (komish) [reply certification/engine/engine.go:97]: The goal here is keeping in line with existing checks which implicitly explain what "true" and "false" mean when reading their variable names..

File: certification/internal/shell/validate_operator_bundle.go
Comment (komish) [reply certification/internal/shell/validate_operator_bundle.go:53]: Er, nevermind. I think I misread this.

File: certification/internal/shell/validate_operator_bundle.go
Comment (bcrochet) [reply certification/internal/shell/validate_operator_bundle.go:47]: It really it just for reporting. We are in no way relying on this output to validate the check. I just wanted to log the messages at the appropriate level. It could just be log.Info for any of them.

File: cli/operatorsdk.go
Comment (bcrochet) [reply cli/operatorsdk.go:32]: Correct. Not configurable by the user. It may not be necessary if we never actually support docker instead of podman. Also, if we move to API calls instead of Exec calls, this would probably be moot.

File: certification/internal/shell/validate_operator_bundle.go
Comment (bcrochet) [reply certification/internal/shell/validate_operator_bundle.go:53]: That's what I did?

File: certification/internal/shell/validate_operator_bundle.go
Comment (bcrochet) [reply certification/internal/shell/validate_operator_bundle.go:9]: Isn't this the correct name? I am changing it to Check from Policy.

File: certification/engine/engine.go
Comment (bcrochet) [reply certification/engine/engine.go:97]: That might be a little too verbose for my tastes... But I definitely see the reason for the comment. This one in particular is not very prescriptive. Let me think on it a bit.

File: certification/internal/shell/validate_operator_bundle.go
Comment (komish) [reply certification/internal/shell/validate_operator_bundle.go:9]: By naming here, I meant the adjustment from ValidateOperatorBundle to something more prescriptive. That said, I'll mark this resolved because changing this in the previous comment would require it to fix this variable reference as well. There's no reason for me to have commented twice here. My apologies!

File: certification/internal/shell/validate_operator_bundle.go
Comment (komish) [reply certification/internal/shell/validate_operator_bundle.go:47]: Definitely agree that the reporting is necessary. Just wary about associating the operator-sdk log levels to ours. I think I would prefer the entire log output from invoked tool just be a single log level on our side (some higher log level). 

Starting to feel like a nit, so I think we can table this and disregard. I'll leave it to your discretion 👍

**Review by jomkz — APPROVED** [review]
/lgtm pending any further comments from @komish

File: certification/internal/shell/validate_operator_bundle.go
Comment (bcrochet) [reply certification/internal/shell/validate_operator_bundle.go:47]: As it isn't there log levels, but actual json output from operator-sdk, I'm ok with this one for the time being.

File: certification/engine/engine.go
Comment (bcrochet) [reply certification/engine/engine.go:97]: I'll stick with this for the moment. Let's at least get the last Policy changed to Check. :)

**Pull request merged**
